File: Assessment_18_06_2019/pangram.cpp
#include<bits/stdc++.h>

using namespace std ;

bool is_pangram(string str) {
    int freq[30] = {0};
    int len = str.size();
    
    for (int i = 0; i < len;i++) {
        freq[str[i] - 'A']++;
    }
    for (int i = 0; i < 26; i++) {
        if(freq[i] == 0)
            return false;
    }

        return true;
}

int main() {

    string str;
    str = "Pack my box with five dozen liquor jugs";
    transform(str.begin(), str.end(), str.begin(), ::toupper);

    bool check = is_pangram(str);
    if(check) {
        cout << "String is pangram\n";
    }
    else {
        cout << "String is not pangram\n";
    }
    return 0;
}

File: Assessment_18_06_2019/pangram.py
import string
letter_count = dict(zip(string.ascii_lowercase, [0]*26))

def is_pangram(s) :
    for ch in s: 
        letter_count[ch] = letter_count.get(ch, 0) + 1
    for count in letter_count :
        if letter_count[count] == 0 :
            return 0

    return 1

s = "Pack my box with five dozen liquor jugs"
s = s.lower() 
if(is_pangram(s)) :
    print("String is Pangram") 
else :
    print("String is not pangram")
